crates/apollo-router/src/http_server_factory.rs

```diff
@@ -17,14 +17,15 @@ use tokio::net::TcpListener;
 /// necessary e.g. when listen address changes.
 #[cfg_attr(test, automock)]
 pub(crate) trait HttpServerFactory {
-    fn create<F>(
+    fn create<Router, Route>(
         &self,
-        graph: Arc<F>,
+        router: Arc<Router>,
         configuration: Arc<Configuration>,
         listener: Option<TcpListener>,
     ) -> Pin<Box<dyn Future<Output = Result<HttpServerHandle, FederatedServerError>> + Send>>
     where
-        F: graphql::Fetcher + 'static;
+        Router: graphql::Router<Route> + 'static,
```

Comment:
why is there a `'static` here? From what I see it's definitely not `'static`

Reply:
:thinking: ...

Just to be clear we understand each other. You do know `'static` in trait boundary is not at all the same than `'static` in lifetimes?

Reply:
I checked and we do a tokio::spawn at some point with that. So it needs to be owned (`'static`). See the `'static` [here](https://docs.rs/async-std/1.6.1/async_std/task/fn.spawn.html)

Reply:
ooof right, my mistake :sweat_smile: